packages/svelte/src/compiler/compile/Component.js

```diff
@@ -199,7 +209,21 @@ export default class Component {
 						.replace(process.cwd(), '')
 						.replace(regex_leading_directory_separator, '')
 				: compile_options.filename);
+
+		// line numbers in stack trace frames are 1-based. source maps are 0-based
 		this.locate = getLocator(this.source, { offsetLine: 1 });
+		// @ts-expect-error - fix the type of CompileOptions.sourcemap
+		const tracer = compile_options.sourcemap ? new TraceMap(compile_options.sourcemap) : undefined;
+		this.meta_locate = (c) => {
+			/** @type {{ line: number, column: number }} */
+			let location = this.locate(c);
```

Comment:
it's confusing to me that we used a 1-based locator for `meta_locate` and then subtract 1. if we just used the default sourcemap offset we wouldn't have to deal with that

we could make `meta_locate` use a separate 0-based locator. would that be more expensive though?

or we could make `locate` be 0-based and then just add 1 when we're printing it out. I think that'd be clearer because sourcemaps are 0-based and then we can just treat everything in the codebase as being 0-based. is `locate` exposed externally though and would that be a breaking change that results in a more difficult API for people needing to print stack traces?

Reply:
This won't change much as the source mapping function works on 1-based lines so we have back and forth either way. I think the way it's now is the most consistent (both locate and meta_locate are 1-based) and you do whatever you need to do on the consumer side (and 1-based is needed more often)